mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -141,50 +165,36 @@ def preprocess_mpas(ds, onlyvars=None, selvals=None, iselvals=None,
     conditions. Hence, a default date offset is chosen to be yearoffset=1849,
     (year 0001 of an 1850 run will correspond with Jan 1st, 1850).
 
-    Note, for use with the timeSeriesStats analysis member fields set
-    timeSeriesStats=True and assign timestr.
-
-    The timestr variable designates the appropriate variable to be used as the
-    unlimited dimension for xarray concatenation.  For MPAS-O
-    timestr='time_avg_daysSinceStartOfSim' and for MPAS-Seaice
-    timestr='timeSeriesStatsMonthly_avg_daysSinceStartOfSim_1'.
+    The data set is assumed to have an array of date strings with variable
+    name given by timestr, typically either 'xtime', or 'xtime_start'.
 
     The onlyvars option reduces the dataset to only include variables in the
     onlyvars list. If onlyvars=None, include all dataset variables.
 
     iselvals and selvals provide index and value-based slicing operations for
     individual datasets prior to their merge via xarray.
-    iselvals is a dictionary, e.g., iselvals = {'nVertLevels': slice(0,3),
-                                                'nCells': cellIDs}
-    selvals is a dictionary, e.g., selvals = {'cellLon': 180.0}
+    iselvals is a dictionary, e.g. iselvals = {'nVertLevels': slice(0, 3),
+                                               'nCells': cellIDs}
+    selvals is a dictionary, e.g. selvals = {'cellLon': 180.0}
 
     Phillip J. Wolfram, Milena Veneziani, Luke van Roekel and Xylar Asay-Davis
-    11/25/2016
+    Last modified: 11/25/2016
     """
 
-    # ensure timestr is specified used when timeSeriesStats=True
-    if timeSeriesStats:
-        if timestr is None:
-            assert False, 'A value for timestr is required, e.g., ' + \
-                    'for MPAS-O: time_avg_daysSinceStartOfSim, and ' + \
-                    'for MPAS-Seaice: ' + \
-                    'timeSeriesStatsMonthly_avg_daysSinceStartOfSim_1'
-
-        # compute shifted datetimes
-        daysSinceStart = ds[timestr]
-        datetimes = [datetime.datetime(yearoffset+1, 1, 1) + x
-                     for x in time_series_stat_time(timestr, daysSinceStart)]
+    if isinstance(timestr, (tuple, list)):
+        # we want to average the two
+        assert(len(timestr) == 2)
+        starts = get_datetimes(ds[timestr[0]], yearoffset)
+        ends = get_datetimes(ds[timestr[1]], yearoffset)
+        datetimes = [starts[i] + (ends[i] - starts[i])/2
+                     for i in range(len(starts))]
```

Comment:
Essentially, I think we could have all of 184-192 inside the `get_datetimes` function.  We don't want to make `preprocess_mpas` more complex.